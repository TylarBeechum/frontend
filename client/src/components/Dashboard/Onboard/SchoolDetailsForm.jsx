import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { MaskedInput, Select, Box, Heading } from 'grommet';

import {
  BaseForm,
  BaseTextInput,
  BaseFormField,
  BaseButton
} from '../../../styles/themes';
import queries from './queries';

const SchoolDetailsForm = ({ history }) => {
  const [input, setInput] = useState({
    name: '',
    taxId: '',
    street1: '',
    street2: '',
    city: '',
    state: '',
    zip: '',
    phone: '',
    type: '',
    url: '',
    userId: localStorage.id
  });

  const states = [
    'AL',
    'AK',
    'AS',
    'AZ',
    'AR',
    'CA',
    'CO',
    'CT',
    'DE',
    'DC',
    'FM',
    'FL',
    'GA',
    'GU',
    'HI',
    'ID',
    'IL',
    'IN',
    'IA',
    'KS',
    'KY',
    'LA',
    'ME',
    'MH',
    'MD',
    'MA',
    'MI',
    'MN',
    'MS',
    'MO',
    'MT',
    'NE',
    'NV',
    'NH',
    'NJ',
    'NM',
    'NY',
    'NC',
    'ND',
    'MP',
    'OH',
    'OK',
    'OR',
    'PW',
    'PA',
    'PR',
    'RI',
    'SC',
    'SD',
    'TN',
    'TX',
    'UT',
    'VT',
    'VI',
    'VA',
    'WA',
    'WV',
    'WI',
    'WY'
  ];

  const handleChanges = e => {
    setInput({
      ...input,
      [e.target.name]: e.target.value
    });
  };

  const handleSubmit = async e => {
    e.preventDefault();
    try {
      await queries.addSchoolDetails(input);
      await queries.addRole({
        id: localStorage.id,
        roleId: 2 // Role of a school is set to always be 2
      });
      history.push('/dashboard');
    } catch (err) {
      console.error(err);
    }
  };

  return (
    <SchoolForm onSubmit={handleSubmit}>
      <Box direction="column">
        <Heading margin="20px 0 0 0" alignSelf="center">
          School Details
        </Heading>
        <SchoolFormField label="Institution">
          <SchoolBaseTextInput
            name="name"
            placeholder="Lambda School"
            onChange={handleChanges}
            value={input.name}
            plain={false}
            required
          />
        </SchoolFormField>
        <SchoolFormField label="Tax Id">
          <SchoolBaseTextInput
            labelText="TaxId"
            name="taxId"
            placeholder="TaxId"
            onChange={handleChanges}
            value={input.taxId}
            required
          />
        </SchoolFormField>
        <SchoolFormField label="Address 1">
          <SchoolBaseTextInput
            name="street1"
            placeholder="123 Fake street"
            onChange={handleChanges}
            value={input.street1}
          />
        </SchoolFormField>
        <SchoolFormField label="Address 2">
          <SchoolBaseTextInput
            name="street2"
            placeholder="Apt B"
            onChange={handleChanges}
            value={input.street2}
          />
        </SchoolFormField>
        <SchoolFormField label="City">
          <SchoolBaseTextInput
            name="city"
            placeholder="San Francisco"
            onChange={handleChanges}
            value={input.city}
          />
        </SchoolFormField>
        <SchoolFormField
          label="State"
          name="state"
          component={Select}
          options={states}
          onChange={({ option }) => setInput({ ...input, state: option })}
          value={input.state}
          placeholder="State"
        />
        <SchoolFormField label="Zip Code">
          <SchoolBaseTextInput
            name="zip"
            placeholder="90210"
            onChange={handleChanges}
            value={input.zip}
          />
        </SchoolFormField>
        <SchoolFormField label="Phone Number">
          <SchoolMaskedInput
            mask={[
              { fixed: '(' },
              {
                length: 3,
                regexp: /^[0-9]{1,3}$/,
                placeholder: 'xxx'
              },
              { fixed: ')' },
              { fixed: ' ' },
              {
                length: 3,
                regexp: /^[0-9]{1,3}$/,
                placeholder: 'xxx'
              },
              { fixed: '-' },
              {
                length: 4,
                regexp: /^[0-9]{1,4}$/,
                placeholder: 'xxxx'
              }
            ]}
            value={input.phone}
            name="phone"
            onChange={handleChanges}
            required
          />
        </SchoolFormField>
        <SchoolFormField label="Type of Institution">
          <SchoolBaseTextInput
            name="type"
            placeholder="University"
            onChange={handleChanges}
            value={input.type}
          />
        </SchoolFormField>
        <SchoolFormField label="Institution Website">
          <SchoolBaseTextInput
            labelText="Institution Website"
            name="url"
            placeholder="ls.dev"
            onChange={handleChanges}
            value={input.url}
            required
          />
        </SchoolFormField>
        <SchoolButton type="submit" primary label="Submit" alignSelf="center" />
      </Box>
    </SchoolForm>
  );
};

SchoolDetailsForm.propTypes = {
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

const SchoolForm = styled(BaseForm)`
  margin: 120px auto 100px;
  background-color: white;
  border: 1px solid #d8d8d8;
  border-radius: 2px;
  max-width: 800px;
  width: 100%;
`;

const SchoolButton = styled(BaseButton)`
  text-align: center;
  margin: 15px 20px 50px;
`;

const SchoolBaseTextInput = styled(BaseTextInput)`
  border: none;
`;

const SchoolFormField = styled(BaseFormField)`
  margin: 20px;
  border-bottom: none;
  input {
    /* margin-bottom: 10px; */
  }
`;

const SchoolMaskedInput = styled(MaskedInput)`
  /* border: ${({ theme }) => theme.global.border}; */
`;

export default SchoolDetailsForm;
