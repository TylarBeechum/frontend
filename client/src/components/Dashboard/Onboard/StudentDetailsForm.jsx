import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { toast } from 'react-toastify';
import { useStateValue } from 'react-conflux';
import { MaskedInput, Select, Box, Heading } from 'grommet';

import {
  BaseForm,
  BaseTextInput,
  BaseFormField,
  BaseButton
} from '../../../styles/themes';
import queries from './queries';
import {
  globalContext,
  ON_BOARD_DETAILS
} from '../../../store/reducers/globalReducer';
import {
  studentContext,
  SET_STUDENT_DATA
} from '../../../store/reducers/studentReducer';
import c from '../../../store/constants';

const SchoolDetailsForm = ({ history }) => {
  const [{ user }, dispatchGlobal] = useStateValue(globalContext);
  const [, dispatchStudent] = useStateValue(studentContext);
  const [input, setInput] = useState({
    firstName: '',
    lastName: '',
    middleName: '',
    street1: '',
    street2: '',
    city: '',
    state: '',
    zip: '',
    phone: '',
    userId: user.id
  });

  const handleChanges = e => {
    setInput({
      ...input,
      [e.target.name]: e.target.value
    });
  };

  const handleSubmit = async e => {
    e.preventDefault();
    try {
      await queries.addRole({
        id: user.id,
        roleId: 3 // Role of a student is set to always be 3
      });
      const details = await queries.addStudentDetail({
        fullName: `${input.firstName} ${input.lastName}`,
        ...input
      });
      dispatchGlobal({
        type: ON_BOARD_DETAILS,
        payload: { ...user, roleId: 3 }
      });
      dispatchStudent({
        type: SET_STUDENT_DATA,
        payload: { ...details }
      });
      localStorage.removeItem('token');
      localStorage.token = details.data.addStudentDetail.token;
      toast.success(`Student Details added succesfully`, {
        className: 'status-ok',
        position: toast.POSITION.BOTTOM_CENTER,
        hideProgressBar: true,
        autoClose: true
      });
      history.push('/dashboard');
    } catch (err) {
      console.error(err);
    }
  };

  return (
    <StudentForm onSubmit={handleSubmit}>
      <Box direction="column">
        <Heading margin="20px 0 0 0" alignSelf="center">
          Student Register
        </Heading>
        <StudentFormField label="First Name">
          <StudentBaseTextInput
            name="firstName"
            placeholder="Jane"
            onChange={handleChanges}
            value={input.firstName}
            plain={false}
            required
          />
        </StudentFormField>
        <StudentFormField label="Middle Name">
          <StudentBaseTextInput
            name="middleName"
            placeholder="Emily"
            onChange={handleChanges}
            value={input.middleName}
            plain={false}
          />
        </StudentFormField>
        <StudentFormField label="Last Name">
          <StudentBaseTextInput
            name="lastName"
            placeholder="Doe"
            onChange={handleChanges}
            value={input.lastName}
            plain={false}
            required
          />
        </StudentFormField>
        <StudentFormField label="Address 1">
          <StudentBaseTextInput
            name="street1"
            placeholder="123 Fake street"
            onChange={handleChanges}
            value={input.street1}
          />
        </StudentFormField>
        <StudentFormField label="Address 2">
          <StudentBaseTextInput
            name="street2"
            placeholder="Apt B"
            onChange={handleChanges}
            value={input.street2}
          />
        </StudentFormField>
        <StudentFormField label="City">
          <StudentBaseTextInput
            name="city"
            placeholder="San Francisco"
            onChange={handleChanges}
            value={input.city}
          />
        </StudentFormField>
        <StudentFormField
          label="State"
          name="state"
          component={Select}
          options={c.states}
          onChange={({ option }) => setInput({ ...input, state: option })}
          value={input.state}
          placeholder="State"
        />
        <StudentFormField label="Zip Code">
          <StudentBaseTextInput
            name="zip"
            placeholder="90210"
            onChange={handleChanges}
            value={input.zip}
          />
        </StudentFormField>
        <StudentFormField label="Phone Number">
          <StudentMaskedInput
            mask={[
              { fixed: '(' },
              {
                length: 3,
                regexp: /^[0-9]{1,3}$/,
                placeholder: 'xxx'
              },
              { fixed: ')' },
              { fixed: ' ' },
              {
                length: 3,
                regexp: /^[0-9]{1,3}$/,
                placeholder: 'xxx'
              },
              { fixed: '-' },
              {
                length: 4,
                regexp: /^[0-9]{1,4}$/,
                placeholder: 'xxxx'
              }
            ]}
            value={input.phone}
            name="phone"
            onChange={handleChanges}
            required
          />
        </StudentFormField>
        <StudentButton
          type="submit"
          primary
          label="Submit"
          alignSelf="center"
        />
      </Box>
    </StudentForm>
  );
};

SchoolDetailsForm.propTypes = {
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

const StudentForm = styled(BaseForm)`
  margin: 120px auto 100px;
  background-color: white;
  border: 1px solid #d8d8d8;
  border-radius: 2px;
  max-width: 800px;
  width: 100%;
`;

const StudentButton = styled(BaseButton)`
  text-align: center;
  margin: 15px 20px 50px;
`;

const StudentBaseTextInput = styled(BaseTextInput)`
  border: none;
`;

const StudentFormField = styled(BaseFormField)`
  margin: 20px;
  border-bottom: none;
  input {
    /* margin-bottom: 10px; */
  }
`;

const StudentMaskedInput = styled(MaskedInput)`
  /* border: ${({ theme }) => theme.global.border}; */
`;

export default SchoolDetailsForm;
