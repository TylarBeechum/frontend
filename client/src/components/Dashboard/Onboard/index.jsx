import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { BaseButton } from '../../../styles/themes';

function Onboard({ history }) {
  return (
    <>
      <Container>
        <div>
          <h2>Choose Account Type</h2>
          <BaseButton
            type="button"
            onClick={() => history.push('/onboarding/school')}
            label="School"
          />
          <BaseButton
            type="button"
            onClick={() => history.push('/onboarding/student')}
            label="Student"
            disabled
          />
        </div>
      </Container>
    </>
  );
}

Onboard.propTypes = {
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

const Container = styled.main`
  width: 100%;
  min-height: 100vh;
  padding: 120px 3% 0;

  div {
    background: ${props => props.theme.global.colors.dashBoardBg};
    border: 1px solid ${props => props.theme.global.colors.dashBoardBorder};
    border-radius: 2px;
    width: 100%;
    max-width: 500px;
    margin: 0 auto;
    padding: 3%;
    display: flex;
    flex-direction: column;
    align-items: center;

    h2 {
      width: 100%;
      text-align: center;
      margin-bottom: 37px;
    }

    button {
      margin-bottom: 25px;
      width: 57%;
      padding: 10px;
      border-radius: 50px;
      :hover {
        background: ${props => props.theme.global.colors.brand};
        color: ${props => props.theme.global.colors['accent-4']};
      }
    }

    button:last-of-type {
      border: 2px solid grey;
      color: grey;
      :hover {
        background: white;
        border: 2px solid grey;
        color: grey;
      }
    }
  }
`;

export default Onboard;
