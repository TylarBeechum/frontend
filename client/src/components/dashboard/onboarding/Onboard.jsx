import React from 'react';
import { Route } from 'react-router-dom';

function Onboard({ history }) {
  const addRole = ({ id, roleId }) => {
    history.push('/onboarding/school');
  };
  return (
    <>
      <button onClick={addRole}>School</button>
    </>
  );
}

export default Onboard;
