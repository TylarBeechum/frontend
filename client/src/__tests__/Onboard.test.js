/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../__testHelpers__';

import Onboard from '../components/Dashboard/Onboard';

describe('<Onboard />', () => {
  describe('should render the CredCard component successfully without crashing', () => {
    it('tests that the component does not crash when it is rendered', () => {
      const helpers = renderWithRouterAndProviders(
        <Onboard history={{ location: { pathname: '/' } }} />
      );
    });
  });

  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const tree = rendererWithRouterAndProviders(
        <Onboard history={{ location: { pathname: '/' } }} />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('should render the required string to the screen successfully', () => {
    it('tests that form title is rendered successfully', () => {
      const { getByText } = renderWithRouterAndProviders(
        <Onboard history={{ location: { pathname: '/' } }} />
      );
      expect(getByText(/Choose Account Type/i)).toBeDefined();
    });
  });

  describe('should fire mock functions when the student and school buttons are clicked', () => {
    it('tests that the school button fires a function when it is clicked', () => {
      const mock = jest.fn();
      const { getByText } = renderWithRouterAndProviders(
        <Onboard history={{ location: { pathname: '/' }, push: mock }} />
      );
      const schoolBtn = getByText(/School/i);
      schoolBtn.click();
      expect(mock).toHaveBeenCalledTimes(1);
    });

    it('tests that the student butotn fires a function when it is clicked', () => {
      const mock = jest.fn();
      const { getByText } = renderWithRouterAndProviders(
        <Onboard history={{ location: { pathname: '/' }, push: mock }} />
      );
      const studentBtn = getByText(/Student/i);
      studentBtn.click();
      studentBtn.click();
      expect(mock).toHaveBeenCalledTimes(2);
    });
  });
});
