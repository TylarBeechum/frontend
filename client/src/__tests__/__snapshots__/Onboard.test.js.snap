// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Onboard /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <main
    className="sc-EHOje bjJUkB"
  >
    <div>
      <h2>
        Choose Account Type
      </h2>
      <button
        className="sc-bdVaJa bDWFJH StyledButton-sc-323bzc-0 kHrfdt"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        School
      </button>
      <button
        className="sc-bdVaJa bDWFJH StyledButton-sc-323bzc-0 kHrfdt"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        Student
      </button>
    </div>
  </main>
</div>
`;
